code/ai/aicode.cpp

```diff
@@ -1791,7 +1793,7 @@ float get_wing_lowest_av_ab_speed(object *objp)
 
 	wingnum = aip->wing;
 
-	if (((shipp->flags[Ship::Ship_Flags::Afterburner_locked]) || !(sip->flags[Ship::Info_Flags::Afterburner])) || (shipp->current_max_speed < 5.0f) || (objp->phys_info.afterburner_max_vel.xyz.z <= shipp->current_max_speed) || !(aip->ai_flags[AI::AI_Flags::Free_afterburner_use]))	{
+	if (((shipp->flags[Ship::Ship_Flags::Afterburner_locked]) || !(sip->flags[Ship::Info_Flags::Afterburner])) || (shipp->current_max_speed < 5.0f) || (objp->phys_info.afterburner_max_vel.xyz.z <= shipp->current_max_speed) || !(aip->ai_flags[AI::AI_Flags::Free_afterburner_use] || aip->ai_profile_flags[AI::Profile_Flags::Free_afterburner_use]))	{
```

Comment:
I think you want to apply the logical-not to both flags separately.  See de Morgan's laws for how the expression is resolved.

Then again, I am not 100% sure I understand the purpose of this function.  Clarification would be appreciated.

Reply:
I'm not completely sure either but I read it as finding the lowest max speed of all ships in a wing.  The `if` clauses you pointed out are to check whether afterburning is enabled.  If not, use the regular max speed.  Otherwise, get the afterburner max speed.
I made the truth tables for what the desired changes (sorry about the formatting but I'm not sure how to fix it; I attached a text file if that's easier to read):

current:
profile		sexp		result
YES			true		false
YES			false		false
NO			true		false
NO			false		true

changed:
profile		sexp		result
YES			true		false
YES			false		true
NO			true		true
NO			false		true

I'm fairly confident that the first one is correct, since you want that condition to be false (i.e. you don't want to limit the speed) in all cases except when both the SEXP and table flags are false.

Reply:
Gotcha.  Now that I've had a chance to look at it when I'm not sleep deprived, I think I can follow the logic, and your explanation is the right thing to do.